engine/src/main/scala/cromwell/webservice/metadata/MetadataBuilderRegulatorActor.scala

```diff
@@ -0,0 +1,57 @@
+package cromwell.webservice.metadata
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import cromwell.core.Dispatcher.ApiDispatcher
+import cromwell.services.metadata.MetadataService.MetadataServiceAction
+import cromwell.webservice.metadata.MetadataBuilderActor.MetadataBuilderActorResponse
+
+import scala.collection.mutable
+
+class MetadataBuilderRegulatorActor(serviceRegistryActor: ActorRef) extends Actor with ActorLogging {
+  // This actor tracks all requests coming in from the API service and spins up new builders as needed to service them.
+  // If the processing of an identical request is already in flight the requester will be added to a set of requesters
+  // to notify when the response from the first request becomes available.
+
+  // Map from requests (MetadataServiceActions) to requesters.
+  val apiRequests = new mutable.HashMap[MetadataServiceAction, Set[ActorRef]]()
+  // Map from ActorRefs of MetadataBuilderActors to requests. When a response comes back from a MetadataBuilderActor its
+  // ActorRef is used as the lookup key in this Map. The result of that lookup yields the request which in turn is used
+  // as the lookup key for requesters in the above Map.
+  val builderRequests = new mutable.HashMap[ActorRef, MetadataServiceAction]()
+
+  override def receive: Receive = {
+    case action: MetadataServiceAction =>
+      val currentRequesters = apiRequests.getOrElse(action, Set.empty)
+      apiRequests.put(action, currentRequesters + sender())
+      if (currentRequesters.isEmpty) {
+        val metadataBuilderActor = context.actorOf(
+          MetadataBuilderActor.props(serviceRegistryActor).withDispatcher(ApiDispatcher), MetadataBuilderActor.uniqueActorName)
+        builderRequests.put(metadataBuilderActor, action)
+        metadataBuilderActor ! action
+      }
+    case response: MetadataBuilderActorResponse =>
+      val sndr = sender()
+      builderRequests.get(sndr) match {
+        case Some(action) =>
+          apiRequests.get(action) match {
+            case Some(requesters) =>
+              apiRequests.remove(action)
+              requesters foreach { _ ! response}
```

Comment:
Metrics-wise, would we be able to record:
- The number of requests being responded to
- How long the oldest requested has been waiting

Reply:
We can certainly open an enhancement issue for those.